c10/core/TensorImpl.cpp

```diff
@@ -307,8 +366,19 @@ SymBool TensorImpl::compute_channels_last_contiguous_2d(
   if (is_sparse()) {
     return false;
   }
-  return _compute_channels_last_contiguous_2d<c10::SymInt>(
-      extra_meta_->sizes_, extra_meta_->strides_);
+  SymIntArrayRef sizes = extra_meta_->sizes_;
+  SymIntArrayRef strides = extra_meta_->strides_;
+  auto n = normalize_sym_sizes_strides(sizes, strides);
+  if (n.has_value()) {
+    SymNode base;
+    std::vector<SymNode> size_nodes;
+    std::vector<SymNode> stride_nodes;
+    std::tie(base, size_nodes, stride_nodes) = *n;
+    return SymBool(
+        base->is_channels_last_contiguous_2d(size_nodes, stride_nodes));
+  } else {
+    return _compute_channels_last_contiguous_2d(sizes, strides);
+  }
```

Comment:
Duplicated above, I feel like a helper method is warranted here, or a template if the contracts between `is_contiguous`/`is_channels_last_contiguous_2d` and `_compute_contiguous`/`_compute_channels_last_contiguous_2d` are similar enough...

Reply:
Woof, just scrolled down, I def think we should template this, you do it 6 times.

Reply:
The primary annoyance with templating is handling the method call. You want to do it with a lambda?

Reply:
lambda runtime is fine, but you can just type it, they all have statically determinable signatures, I think https://godbolt.org/z/4b8nKhxP8 should work?

Reply:
Want me to commandeer and make the change? It's just boilerplate management at this point, would be happy to.

Reply:
Discussed offline, these are methods, so https://godbolt.org/z/6The79rzd maybe?

Reply:
Just passing private method pointers isn't enough, because the code you want to template also access private attributes on TensorImpl. So we'd need at least two separate types referenced here.

(monkey's paw curls)

congrats! you're getting a macro'ified version of this method instead :P